Bugsnag/Client/BugsnagClient.m

```diff
@@ -715,14 +739,96 @@ - (BugsnagCorrelation *)getCurrentCorrelation {
     return [[BugsnagCorrelation alloc] initWithTraceId:traceId spanId:spanId];
 }
 
+- (BugsnagMetadata *)metadataCapture:(BugsnagCaptureOptions *)capture
+               metadata:(BugsnagMetadata *)metadata
+              unhandled:(BOOL)unhandled {
+    if (unhandled == TRUE || capture == nil || capture.metadata == nil) {
+        // Copy all of the current metadata from self
+        for (NSString* sectionKey in self.metadata.dictionary) {
+            if (sectionKey == nil) {
+                continue;
+            }
+            NSMutableDictionary *metadataInSection = [self.metadata getMetadataFromSection:sectionKey];
+            if (metadataInSection != nil) {
+                [metadata addMetadata:metadataInSection toSection:sectionKey];
+            }
+        }
+        return metadata;
```

Comment:
Could you use `[self.metadata copy]` for this case?

Reply:
no, because we enter this method with potential metadata from nsexception/nserror processing and calling `copy` would override that.